tests/tests.go

```diff
@@ -32,14 +33,22 @@ func StartServer(t testing.TB, cfg *serverconfig.Config) {
 	StartServerWithContext(t, cfg, serverCtx)
 }
 
-// StartServerWithContext starts a server with a specific ServerContext and waits until it is healthy.
+// StartServerWithContext starts a server in random ports and with a specific ServerContext and waits until it is healthy.
 // When the test ends, all resources are cleaned.
 func StartServerWithContext(t testing.TB, cfg *serverconfig.Config, serverCtx *run.ServerContext) {
 	container := storage.RunDatastoreTestContainer(t, cfg.Datastore.Engine)
 	cfg.Datastore.URI = container.GetConnectionURI(true)
 
 	ctx, cancel := context.WithCancel(context.Background())
 
+	httpPort, httpPortReleaser := testutils.TCPRandomPort()
```

Comment:
What's the hypothesis for putting this here?

The port is bound as soon as we call `TCPRandomPort()` earlier in the call chain (at the time we construct the config with `MustDefautlconfigWithRandomPorts`).

Reply:
> The port is bound as soon as we call TCPRandomPort() 

yes and no.. `TCPRandomPort` finds a free port and then liberates with the function returned. If there is a long delay between the port liberation and the port acquiring by the server, we see the following: (below i am copy-pasting what I put in the PR description):

I believe the root cause of this error is that on main we are doing this when spinning up a server for testing:

1. ask for a random TCP port "x"
2. free "x"
3. spin up a test container (which can take time)
4. start the server on port "x"

which can occasionally produce this:

1. test A ask for a random TCP port "x"
2. test A frees "x"
3. test A spins up a test container (which can take time)
4. test B, in parallel, asks for a random TCP port and also gets "x"
5. test B spins up the test container quickly and starts the server on port "x"
6. test A tries to start the server on port "x" and gets bind: address already in use

Reply:
What's not clear to me is how is a port free without it having been liberated though? I wouldn't expect the OS to be returning a "free" port if it is not yet liberated (e.g. the closer has been called on the port).